src/doc/src/reference/environment-variables.md

```diff
@@ -265,7 +265,7 @@ corresponding environment variable is set to the empty string, `""`.
   where integration tests or benchmarks are free to put any data needed by
   the tests/benches. Cargo initially creates this directory but doesn't
   manage its content in any way, this is the responsibility of the test code.
-* `CARGO_RUSTC_CURRENT_DIR` --- This is a path that `rustc` is invoked from **(nightly only)**.
+* `CARGO_RUSTC_CURRENT_DIR` --- This is a path that `rustc` is invoked from
```

Comment:
Still feel like people might overlook this, as it isn't an obvious thing easy to relate to their problems. That said, I am fine with the previous explanation. Just forward it here if people have the same question: <https://github.com/rust-lang/cargo/pull/12996#discussion_r1404504001>.

> When we talked about this, Eric was hesitant about locking down the behavior of `file!` so I left out that use case. It should work and I can't imagine rustc doing anything but absolute or relative to `std::env::current_dir` but I'm leaving that to people to make the choice that they want to make those assumptions, rather than us advertising it.